### ISSUE TITLE
Automatic geometry simplification is too aggressive

### ISSUE THREAD

**Issue by matkoniecz** [openstreetmap/iD#8364]
While I was mapping meandering river using iD ( https://www.openstreetmap.org/way/910390742#map=15/31.7363/-89.1748 ), I noticed that iD is automatically removing some of placed nodes.

It resulted in a noticeably worse geometry.

Would it be possible to reduce aggressiveness of that mechanism? I can attach screencast if that would be useful.

**Comment by DujaOSM** [comment]
I'm not a contributor, but I'm curious what you refer to as "automatic geometry simplification"; I'm not aware that iD has such stuff. 

Just for fun, I continued your river (it's apparently [Tallahoma Creek](https://naturalatlas.com/creeks/tallahoma-954402)). iD has no "curve smoothing" that I know of, so I had to click once for every node in the way, and it kept all my clicks (save for a missed odd one). So I guess a screencast would be useful.

**Comment by matkoniecz** [comment]
![Peek 2021-02-24 12-53](https://user-images.githubusercontent.com/899988/108996951-57368300-769f-11eb-90c6-d975fb0a767f.gif)

On trying again it seems that it happens when I am quickly pressing buttons so maybe

- button click is never triggered and I am highly confused person unable to use mouse (PEBKAC, but it is not happening to me in JOSM)
- button click is not registered due to some performance issues
- iD discards click close to each other / quick after another

In red: click that was missed
![screenshot-b8c21ffd](https://user-images.githubusercontent.com/899988/108997675-2d319080-76a0-11eb-8b5c-219b3ea7755c.png)

So: is there something in iD what discards clicks? Or should I debug whatever click is even triggered?

**Comment by matkoniecz** [comment]
> Just for fun, I continued your river (it's apparently Tallahoma Creek).

Thanks! I continued it a bit upstream, https://www.openstreetmap.org/note/2552383 waits for mapper who would want to continue.

**Comment by DujaOSM** [comment]
Yes, that also happens to me occasionally (thus my remark "save for a missed odd one"). However, it's not too often so I barely notice, and there's (usually) visual feedback so I can return and fix it... 

iD does queue clicks so, even when it becomes sluggishly unresponsive (#8091 ) it will render newly drawn objects, even with several seconds delay. Not that I'm happy when #8091 kicks in...

**Comment by quincylvania** [comment]
@matkoniecz From your screen capture, it looks like the dropped node may have occurred because you dragged the mouse somewhat, so iD interpreted the action as panning the map instead of clicking in one place. There is already a small tolerance so you don't have to release the mouse in exactly the same spot, but perhaps it needs to be increased?

**Comment by matkoniecz** [comment]
> because you dragged the mouse somewhat, so iD interpreted the action as panning the map instead of clicking in one place

That would definitely fit! I was quickly moving + clicking so it is possible that both were combined.

Is it possible that iD is more affected by such combining due to some lags or is it getting the same mouse input data as any other software? (Firefox on Ubuntu, if that matters - I can also test it on other browsers and on Windows if that would be useful)

> but perhaps it needs to be increased?

Do you remember where it is defined? Tweaking parameter seems like a PR that I can do without spending time on understanding general architecture, while being time-consuming to test. So ideal for a drive-by PR.

**Comment by quincylvania** [comment]
The variable is set [here](https://github.com/openstreetmap/iD/blob/8a9b4d53210df3a06f7e4acc46e3497eb574d286/modules/behavior/draw.js#L30). The tolerance seems reasonable on my machine, but I know different hardware/software setups may see different results.

**Comment by matkoniecz** [comment]
OK, so there is both geometry simplification and blocking quick node additions if I read code correctly ( https://github.com/openstreetmap/iD/blob/8a9b4d53210df3a06f7e4acc46e3497eb574d286/modules/behavior/draw.js#L92 - is it OK to assume that distance is in meters? It seems to be discarding cases where distance is smaller than 4 in some units and when last click was less than 0.5s ago).

And there is risk that tweaking them to fit my preferences may reduce fit for someone else - do you know what was the motivation for that? Preventing accidental double clicks?


BWT, link in https://github.com/openstreetmap/iD/blob/8a9b4d53210df3a06f7e4acc46e3497eb574d286/modules/geo/vector.js#L38 is dead: https://jsperf.com/id-dist-optimization throws 500 error

**Comment by matkoniecz** [comment]
not a problem I noticed recently
